### PR TITLE
Generic mosaic config to support any EIP20 Token

### LINKED ISSUE #141: Define generic mosaic config
- Make the keys in mosaic-config generic, so that it can represent any ERC20 token.
- Rename the keys
   - simpleTokenAddress -> valueTokenAddress
   - ostComposerAddress -> stakePool
   - ostGatewayOrganizationAddress -> gatewayOrganizationAddress 
   - ostEIP20GatewayAddress -> eip20GatewayAddress
   - ostCoGatewayOrganizationAddress -> coGatewayOrganizationAddress
   - ostPrimeAddress -> utilityTokenAddress
   - ostEIP20CogatewayAddress -> eip20CogatewayAddress
- Make the following keys optional
  - bootNodes
  - genesis

Update the code in mosaic-chains and facilitator to use these new keys.
- Also add below fields in mosaic config
     - baseToken in origin
     - baseToken in aux

### PR DESCRIPTION
**Pull request by abhayks1**
PR covers below:

- Mosaic config keys are more generic to support EIP20 token.
- baseToken key has been added on both origin and aux object.

**Note**:  We need to update mosaic config of docker for dev chain and republish it again. 
  Travis tests will pass after that.

Fixes #141

### CODE REVIEW

**Comment by puneet-khushwani-eth** [comment]
starting to review this 👀

File: src/Graph/SubGraph.ts [hunk 1]
```diff
@@ -193,8 +193,9 @@ export default class SubGraph {
     const mosaicConfig: MosaicConfig = MosaicConfig.fromChain(this.originChain);
     return {
       projectRoot: Directory.projectRoot,
-      ostComposerAddress: mosaicConfig.originChain.contractAddresses.ostComposerAddress,
-      eip20GatewayAddress: mosaicConfig.auxiliaryChains[this.auxiliaryChain].contractAddresses.origin.ostEIP20GatewayAddress,
+      stakePoolAddress: mosaicConfig.originChain.contractAddresses.stakePoolAddress,
+      baseTokenAddress: mosaicConfig.originChain.contractAddresses.valueTokenAddress,
```
Comment (puneet-khushwani-eth) [inline src/Graph/SubGraph.ts]: why do we need `baseTokenAddress ` here ? I guess it is not being used in sub graph deployment

File: package-lock.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@openst/mosaic-chains",
-  "version": "0.1.0-alpha.1",
+  "version": "0.1.0-alpha.2",
```
Comment (puneet-khushwani-eth) [inline package-lock.json]:
```suggestion
  "version": "0.1.0-alpha.5",
```

Comment (puneet-khushwani-eth) [reply package-lock.json]: also please change package version in package.json to `0.1.0-alpha.5`

**Comment by puneet-khushwani-eth** [comment]
Also as https://github.com/mosaicdao/mosaic-chains/pull/120 is merged, please make the necessary changes in `mosaic.config` for`goerli`

File: docker/mosaic.json [hunk 3]
```diff
@@ -16,17 +16,19 @@
...
           "anchorAddress": "0xEa8D41fc6C6C0Ee155E5F6FbF9Cc1167Fa17927E",
-          "ostGatewayOrganizationAddress": "0xaB5335771fBA0270634e649882229c7e8CaA6158",
-          "ostEIP20GatewayAddress": "0xA7f056b1320fE619571849f138Cd1Ae2f2e64179"
+          "gatewayOrganizationAddress": "0xaB5335771fBA0270634e649882229c7e8CaA6158",
+          "eip20GatewayAddress": "0xA7f056b1320fE619571849f138Cd1Ae2f2e64179"
         },
         "auxiliary": {
           "anchorOrganizationAddress": "0xb8c982d0cd5f7c1B24B80f0e38cB25a6d9b5b716",
           "anchorAddress": "0x50c0Af1A754CE6bA25a102D4ee759b08141C4aA9",
           "ostCoGatewayOrganizationAddress": "0xbc67Ff0116248762c940B97bA1501BeD6Fa6D093",
```
[the remark this answers is not in the document]
Comment (abhayks1) [reply docker/mosaic.json]: Done!

File: package-lock.json (hunk 2 above)
Comment (abhayks1) [reply package-lock.json]: Done!

File: src/Graph/SubGraph.ts (hunk 1 above)
Comment (abhayks1) [reply src/Graph/SubGraph.ts]: Agree! reverted the change.

File: docker/mosaic.json [hunk 4]
```diff
@@ -4,8 +4,8 @@
       "gatewayLibAddress": "0x353AEba14b902e67cDdA99595f74d910e3FD5e97",
       "messageBusAddress": "0xB69f2Aa06793028CC8E604122218b1D2E23eae4b",
       "merklePatriciaLibAddress": "0xf8238D9850F2a47f1411Cb7F5F5F3279A72b7f7a",
```
Comment (puneet-khushwani-eth) [inline docker/mosaic.json]: also please run mosaic-verify for all these mosaic-configs

Comment (puneet-khushwani-eth) [reply docker/mosaic.json]: looks like mosaic-verify does not take `config-path` as param and always defaults to verifying config for baseToken.

Comment (puneet-khushwani-eth) [reply docker/mosaic.json]: We might have to start taking as CLI input

Comment (abhayks1) [reply docker/mosaic.json]: We have a ticket for this: https://github.com/mosaicdao/mosaic-chains/issues/94
We can verify with above ticket.

File: src/Config/MosaicConfig.schema.json [hunk 5]
```diff
@@ -209,10 +214,8 @@
         }
       },
       "required": [
-        "bootNodes",
```
Comment (sarvesh-ost) [inline src/Config/MosaicConfig.schema.json]: Bootnodes and genesis must be required params.

File: src/NewChain/ChainVerifier.ts [hunk 6]
```diff
@@ -143,24 +143,17 @@ export default class ChainVerifier {
...
 
     await this.validateBIN(
       this.auxiliaryWeb3,
       'Organization',
-      this.contractAddresses.auxiliary.ostCoGatewayOrganizationAddress,
+      this.contractAddresses.auxiliary.coGatewayOrganizationAddress,
       'ContractsBin: Mismatch of auxiliary coGateway organization BIN!!!',
     );
 
-    await this.validateBIN(
```
Comment (sarvesh-ost) [inline src/NewChain/ChainVerifier.ts]: Why this is removed?

File: src/NewChain/ChainVerifier.ts [hunk 7]
```diff
@@ -185,31 +178,23 @@ export default class ChainVerifier {
     ChainVerifier.validateDeployedAddress(
       this.originWeb3,
       valueToken,
-      this.mosaicConfig.originChain.contractAddresses.simpleTokenAddress,
+      this.mosaicConfig.originChain.contractAddresses.valueTokenAddress,
       'Gateway: Invalid valueToken address!!!',
     );
 
-    const baseToken = await gatewayInstance.methods.baseToken().call();
```
Comment (sarvesh-ost) [inline src/NewChain/ChainVerifier.ts]: Why this is removed?

**Pull request merged**